app/controllers/devise/saml_sessions_controller.rb

```diff
@@ -17,6 +17,17 @@ def metadata
     render :xml => meta.generate(@saml_config)
   end
 
+  def idp_sign_out
+    if params[:SAMLRequest] && Devise.saml_session_index_key
+      logout_request = OneLogin::RubySaml::SloLogoutrequest.new(params[:SAMLRequest], @saml_config)
+      resource_class.reset_session_key_for(logout_request.name_id)
```

Comment:
This should only use `name_id` if `use_subject` is configured, and otherwise use attributes like the model does.